torch/ao/quantization/experimental/fake_quantize.py

```diff
@@ -0,0 +1,35 @@
+import torch
+from torch import Tensor
+from torch.ao.quantization.experimental.observer import APoTObserver
+from torch.ao.quantization.experimental.quantizer import quantize_APoT, dequantize_APoT
+from torch.ao.quantization.fake_quantize import FakeQuantizeBase
+
+class APoTFakeQuantize(FakeQuantizeBase):
+    alpha: Tensor
+    gamma: Tensor
+    quantization_levels: Tensor
+    level_indices: Tensor
+
+    def __init__(self, observer: APoTObserver):
+        super().__init__()
+        self.activation_post_process = observer
+        self.alpha = self.gamma = self.quantization_levels = self.level_indices = Tensor()
```

Comment:
why is this initialized to empty tensor instead of attriubtes in observer? this makes the forward method faulty

Reply:
@dzdang i initialize these qparams in forward by calling `calculate_qparams` , like https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fake_quantize.py#L158 (here the qparams are not set in init). should i be calling `calculate_qparams` in my constructor instead?

Reply:
what if `if self.observer_enabled[0] == 1`? does your workflow guaranteed the qparams are set before you come across the case self.observer_enabled[0] == 1?

Reply:
@dzdang In the case `self.fake_quant_enabled[0] == 1` , I have an assert statement to check that the qparams have been set before I use the qparams.
I wasn't sure how to initialize elsewhere besides in the `self.observer_enabled[0] == 1` case in `forward` since if observer is not enabled shouldn't I not be calling `observer.calculate_qparam` to set the qparams? I thought this made most sense and was most consistent with the uniform fake_quantize but lmk what you think

Reply:
this is related to the constructor args comment. if you create an observer in the constructor, then couldn't you just assign the fake parameter's qparams to the observer's?

Reply:
do you mean assigning qparams what is returned by `observer.calculate_qparams` ? observer does not store qparams, just `b`, `k`, `n`, `max_val`, `min_val`. calculate_qparams also takes an additional arg, `signed`

Reply:
oh right. hmm I think what you've done is fine then.